plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Library.java

```diff
@@ -102,4 +104,14 @@ public void setLibraryDependencies(List<String> libraryDependencies) {
       this.libraryDependencies = new ArrayList<>(libraryDependencies);
     }
   }
+
+  public void setRecommendation(LibraryRecommendation recommendation) {
+    if (recommendation != null) {
+      this.recommendation = recommendation;
```

Comment:
The response I can imagine would be to throw an exception. My feeling is that the intention of clients setting this to null is the same as setting it to an empty collection -> clearing it.